Added proper tab completion for /nation

stzups opened the pull request:
<!--- Welcome! It looks like you're opening a pull request for the Towny project, we think that's great. This form is pre-populated with a Contributor License Agreement, which is required if you want to contribute your code. It is there to protect your copyright over the code but also to protect Towny, making your code available to us to use indefinitely. --->
#### Description: 
<!--- Describe your Pull Request's purpose here please. --->
This pull request adds proper /nation tab complete support. Some more important changes are the returning of an `Collections.emptyList()` instead of `null` if there are no valid tab completes. `null` simply returns online players, and can lead to irrelevant results. 

Additionally, the names of nations will only be returned if the player is not typing any valid sub commands. For example, typing  `/nation e` will only return sub commands such as `enemylist`, If `/nation ev` is typed, nation names will be returned such as an example nation called `Everett`. This reduces clutter and reduces the times where all nation names are checked.

This pull request also adds more methods to `NameUtil.java`. These methods are used for getting the resident names of a player's town or nation, and getting the town names of a player's nation. These method names are currently long and hard to read. They will be used by other Towny commands such as `/town rank add <resident>`, which should only return the residents of that player's town.

More specific changes include the support of `/n ally add -[nation]` or `/n ally add [nation]`, where the `-` at the beginning still returns a nation name
____
#### New Nodes/Commands/ConfigOptions: 
<!--- If your PR includes any new permission nodes, commands or config options list them here. --->


____
#### Relevant Towny Issue ticket:
<!--- If your pull request addresses an Issue ticket please provide the link to that --->


____
- [x] I have tested this pull request for defects on a server. 
<!--- Place x between [ ] if you have tested this code on a server. --->

By making this pull request, I represent that I have the right to waive copyright and related rights to my contribution, and agree that all copyright and related rights in my contributions are waived, and I acknowledge that the TownyAdvanced organization has the copyright to use and modify my contribution under the Towny [License](https://github.com/LlmDl/Towny/blob/master/LICENSE.md) for perpetuity.

stzups commented:
The `/nation toggle` help page includes `peaceful`, `public`, and `open`, but `neutral` is also a valid substitute for `peaceful`. Should this be included in the tab completion for `/nation toggle`?

LlmDl commented on src/com/palmergames/bukkit/towny/command/NationCommand.java:
```diff
@@ -188,43 +204,110 @@
+					switch (args[0].toLowerCase()) {
+						case "set":
+							return NameUtil.filterByStart(nationSetTabCompletes, args[1]);
+						case "rank":
```
The rank will have to return a list of players in the nation

stzups replied:
It does farther down, see the following [lines](https://github.com/TownyAdvanced/Towny/pull/3710/files/202c0e6584096a036b9cf6468cc4450f1812b726#diff-7e716e72bf8f5bffab172ac2a81e7a83R265-R276) around 270-ish. 

I'm not sure if I want to go with this style, but for the first `switch` statement I have cases that return possible sub commands. If none of the cases return anything, the second `switch` inside of the `if (args.length > 1) {` will handle returning possible residents, towns, nations, ranks, perms, etc. 

I may want to everything in one `switch` which has cases for each sub command. This may be easier in terms of readability with especially long commands, such as `/t set perm ally build off` as an example

LlmDl commented on src/com/palmergames/bukkit/towny/command/NationCommand.java:
```diff
@@ -188,43 +204,110 @@
+							return NameUtil.filterByStart(nationAllyTabCompletes, args[1]);
+						case "toggle":
+							return NameUtil.filterByStart(nationToggleTabCompletes, args[1]);
+						case "king":
```
Not sure what the king bit here is for, /n set king is a command but I don't think /n king is.

stzups replied:
`/n king` is not a command but `/n king ?` shows a help page for commands relevant to a nation's king. This isn't necessary but is on purpose.

LlmDl replied:
Ah right

LlmDl commented on src/com/palmergames/bukkit/towny/command/NationCommand.java:
```diff
@@ -188,43 +204,110 @@
+					case "delete":
+					case "spawn":
+						return NameUtil.getNationNamesStartingWith(args[1]);
+					case "new":
```
I believe this one you're handling the admin ability to make a nation for a capital, this command has already been removed (and will be off the commands wikipage after we put out the  next release build.)

stzups replied:
Removed

LlmDl commented on src/com/palmergames/bukkit/towny/command/NationCommand.java:
```diff
@@ -110,10 +111,25 @@
-	public static final List<String> nationToggleTabCompletes = new ArrayList<>(Arrays.asList(
-		"neutral",
+	static final List<String> nationToggleTabCompletes = new ArrayList<>(Arrays.asList(
+		"peaceful",
```
The original command was neutral, and someone thought this wasn't quite clear enough and changed it to peaceful. Both should be available.

stzups replied:
If both are available, then should both be show in the `/n toggle` help? This may be confusing to users because there are two different words that do the same thing. What if the `/n toggle` looked like this?
![image](https://user-images.githubusercontent.com/55036602/74989789-6a20d280-540f-11ea-9b86-4a0b3d767733.png)

LlmDl replied:
make it so

stzups commented:
Latest changes include: 

- Moved everything to one `switch`
- Added `king` to `nationTabCompletes` (`/n king`)
- Made console help static
- `/n ally add -[nation]` only returns sent invites to be able to revoke them
- `/n ally remove [nation]` only returns current allies

stzups commented:
That should be it for now in terms of features with the nation command. Thanks for your feedback and I hope this format works consistently with other commands.
